### PR TITLE
Feature: Support Scene Lora with SDXL

### PR DESCRIPTION
**Pull request by hkunzhe**
1. Support Scene Lora with SDXL.
2. User can upload caption files with training photos to train the scene lora.
image

3. Use original training photos instead of cropping if `crop_ratio < 1`.
4. Add sanity check.
5. Add the logger for `easyphoto_train.py`.

### CODE REVIEW

File: scripts/easyphoto_train.py [hunk 1]
```diff
@@ -95,18 +104,21 @@ def easyphoto_train_forward(
 
     # check if user want to train Scene Lora
     train_scene_lora_bool = True if train_mode_choose == "Train Scene Lora" else False
+    if train_scene_lora_bool and float(crop_ratio) < 1:
```
Comment (bubbliiiing) [inline scripts/easyphoto_train.py:107]: 要不要改成-1哥，一般感觉不设置都是-1

File: scripts/easyphoto_utils/common_utils.py [hunk 2]
```diff
@@ -424,15 +424,12 @@
...
+    from scripts.sdwebui import get_scene_prompt
+    
     safetensors_lora_path = os.path.join(models_path, "Lora", lora_path)
     if not safetensors_lora_path.endswith("safetensors"):
         return False
-    lora_type = get_lora_type(safetensors_lora_path)
-    if lora_type == 4:
-        return True
-    return False
+    return get_scene_prompt(safetensors_lora_path)[0]
```
Comment (bubbliiiing) [inline scripts/easyphoto_utils/common_utils.py:432]: 哥，这样写有点疑惑，还是用原来的吧

Comment (bubbliiiing) [reply scripts/easyphoto_utils/common_utils.py:432]: 我看到你改了get_scene_prompt，但很难理解

File: scripts/preprocess.py [hunk 3]
```diff
@@ -250,6 +254,8 @@ def compare_jpg_with_face_id(embedding_list):
     else:
         # jpg list
         jpgs = os.listdir(inputs_dir)
+        # sort photo files to correspond with captions.
+        jpgs = sorted(jpgs, key=lambda x: int(os.path.splitext(os.path.basename(x))[0]))
```
Comment (bubbliiiing) [inline scripts/preprocess.py:258]: txt啥时候sort过吗

**Comment by zouxinyi0625** [comment]
tested. The scene XL LoRA model is trained ~30 minutes (A10).
EasyPhoto-0003

**Review by bubbliiiing — APPROVED** [review]

**Pull request merged**